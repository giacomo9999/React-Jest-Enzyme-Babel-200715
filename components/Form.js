import React from "react";

const Form = ({ handleChange, value, handleClick }) => {
  return (
    <div className="form">
      <input
        className="form-control"
        type="text"
        onChange={handleChange}
        value={value}
      />
      <button className="btn" onClick={handleClick}></button>
    </div>
  );
};

export default Form;
