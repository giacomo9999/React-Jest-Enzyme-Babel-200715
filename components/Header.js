import React from "react";

const Header = ({ message, handleToggleTheme, theme }) => {
  return (
    <div className="header">
      <h1>{message}</h1>
      <button className="right" onClick={handleToggleTheme}>
        <i className={theme}></i>
      </button>
    </div>
  );
};

export default Header;
