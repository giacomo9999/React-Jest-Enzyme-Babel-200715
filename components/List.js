import React from "react";

const List = ({ items }) => {
  return (
    <ul className="list">
      {items.map((item) => (
        <li className="item" key={item}>
          {item}
        </li>
      ))}
    </ul>
  );
};

export default List;
