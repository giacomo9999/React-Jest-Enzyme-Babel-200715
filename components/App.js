import React, { Component } from "react";
import Header from "./Header";
import List from "./List";
import Form from "./Form.js";

class App extends Component {
  state = { listItem: "", items: [], isDarkTheme: false };

  handleChange = ({ target: { value } }) => {
    this.setState({ listItem: value });
  };

  handleClick = () => {
    this.setState((state) => ({
      items: [...state.items, this.state.listItem],
      listItem: "",
    }));
  };

  handleToggleTheme = () => {
    this.setState({ isDarkTheme: !this.state.isDarkTheme });
  };

  render() {
    const theme = this.state.isDarkTheme ? "dark" : "light";
    return (
      <div className={theme}>
        <Header
          theme={theme}
          message={this.props.message}
          handleToggleTheme={this.handleToggleTheme}
        />
        <Form
          handleChange={this.handleChange}
          value={this.state.listItem}
          handleClick={this.handleClick}
        />
        <List items={this.state.items} />
      </div>
    );
  }
}

App.defaultProps = { message: "Hello World" };

export default App;
