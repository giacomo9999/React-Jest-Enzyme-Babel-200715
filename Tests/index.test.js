import React from "react";
import App from "../components/App";
import Header from "../components/Header";
import Form from "../components/Form";
import List from "../components/List";

import { shallow, mount } from "enzyme";

describe("Test App component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />);
  });
  it("Should not return an error", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
