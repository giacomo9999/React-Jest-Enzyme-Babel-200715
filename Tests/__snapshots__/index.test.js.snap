// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Should not return an error 1`] = `ShallowWrapper {}`;
